src/prefect/cli/work_pool.py

```diff
@@ -408,7 +414,12 @@ async def get_available_work_pool_types() -> Set[str]:
         for collection in worker_metadata.values():
             for worker in collection.values():
                 work_pool_types.append(worker.get("type"))
-    except Exception:
+    except Exception as exc:
+        if PREFECT_DEBUG_MODE.value():
+            logger = get_logger()
+            logger.warning(
+                f"Unable to get worker metadata from the collections registry: {exc}"
+            )
```

Comment:
Nice :)

You may want to do `logger.warning(...., exc_info=True)` instead of just including the exception string — especially with debug mode on it seems reasonable to log the traceback.

fun fact: You don't need to call `value()` to check if a bool setting is set :)